crates/transport_iroh/src/lib.rs

```diff
@@ -307,10 +327,16 @@ impl IrohTransport {
         // streams from the remote.
         spawn_connection_reader(ctx.clone(), connections.clone());
 
-        // Perform preflight as first message on the new connection.
+        // Send peer URL to remote first.
+        if let Some(local_url) = local_url {
+            let payload = Bytes::copy_from_slice(local_url.as_str().as_bytes());
+            send_frame(&conn, FrameType::PeerUrl, payload).await?;
+            tokio::time::sleep(Duration::from_millis(10)).await;
```

Comment:
This and the following sleeps are crutches while message order isn't guaranteed yet. That'll be fixed with yet another modification =)

Reply:
There is an issue for it, is that good enough for you? https://github.com/holochain/kitsune2/issues/403

Reply:
Anyway, I added the TODOs.

Reply:
https://github.com/holochain/kitsune2/pull/394/commits/7a85080f1f80c3d696c89648499d0eb2fbdac9a9